packages/@aws-cdk/aws-codebuild/lib/source.ts

```diff
@@ -793,6 +859,10 @@ class BitBucketSource extends ThirdPartyGitSource {
       sourceProperty: {
         ...superConfig.sourceProperty,
         location: this.httpsCloneUrl,
+        buildStatusConfig: {
+          context: this.buildStatusName,
+          targetUrl: this.buildStatusUrl,
+        },
```

Comment:
Same here:

We need to render this property only if at least one of `this.buildStatusContext` or `this.buildStatusUrl` is defined:

```suggestion
        buildStatusConfig: this.buildStatusName !== undefined || this.buildStatusUrl !== undefined
          ? {
            context: this.buildStatusName,
            targetUrl: this.buildStatusUrl,
          }
          : undefined,
```